### ISSUE TITLE
Could not find a suitable method to bind; ember-source ~3.10.0

### ISSUE THREAD

**Issue by wilkosz** [emberjs/ember.js#18136]
## Issue 
Using `ember-cli-uglify` to minify `vendor.js` on `"ember-source": "~3.10.0",` with compress option true is breaking. 

<img width="1636" alt="Screen Shot 2019-06-24 at 12 32 29 PM" src="https://user-images.githubusercontent.com/2351142/60036118-7421e400-967c-11e9-815d-491664d2c859.png">

```json
{
  "name": "test-uglify",
  "version": "0.0.0",
  "private": true,
  "description": "Small description for test-uglify goes here",
  "repository": "",
  "license": "MIT",
  "author": "",
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "build": "ember build",
    "lint:hbs": "ember-template-lint .",
    "lint:js": "eslint .",
    "start": "ember serve",
    "test": "ember test"
  },
  "devDependencies": {
    "@ember/jquery": "^0.6.0",
    "@ember/optional-features": "^0.7.0",
    "broccoli-asset-rev": "^3.0.0",
    "ember-ajax": "^5.0.0",
    "ember-cli": "~3.10.1",
    "ember-cli-app-version": "^3.2.0",
    "ember-cli-babel": "^7.7.3",
    "ember-cli-dependency-checker": "^3.1.0",
    "ember-cli-eslint": "^5.1.0",
    "ember-cli-htmlbars": "^3.0.1",
    "ember-cli-htmlbars-inline-precompile": "^2.1.0",
    "ember-cli-inject-live-reload": "^1.8.2",
    "ember-cli-sri": "^2.1.1",
    "ember-cli-template-lint": "^1.0.0-beta.1",
    "ember-cli-uglify": "^2.1.0",
    "ember-data": "~3.10.0",
    "ember-export-application-global": "^2.0.0",
    "ember-load-initializers": "^2.0.0",
    "ember-maybe-import-regenerator": "^0.1.6",
    "ember-qunit": "^4.4.1",
    "ember-resolver": "^5.0.1",
    "ember-source": "~3.10.0",
    "ember-welcome-page": "^4.0.0",
    "eslint-plugin-ember": "^6.2.0",
    "eslint-plugin-node": "^9.0.1",
    "loader.js": "^4.7.0",
    "qunit-dom": "^0.8.4"
  },
  "engines": {
    "node": "8.* || >= 10.*"
  }
}
```

```js
'use strict';

const EmberApp = require('ember-cli/lib/broccoli/ember-app');

module.exports = function(defaults) {
  let app = new EmberApp(defaults, {
    'ember-cli-uglify': {
      enabled: true,
      uglify: {
        compress: true,
        mangle: true,
        output: {
          semicolons: true,
        },
      },
    },
    sourcemaps: {
      enabled: true,
      extensions: ['js'],
    },
  });
  return app.toTree();
};
```

## Replicate 
- `ember new ember-quickstart`
- add uglify options to `ember-cli-build.js`: 
```json
'ember-cli-uglify': {
      enabled: true,
      uglify: {
        compress: true,
        mangle: true,
        output: {
          semicolons: true,
        },
      },
    },
```

**Comment by locks** [comment]
Thanks for reporting. Could not reproduce in 3.12.

**Comment by dguettler** [comment]
@wilkosz did you ever figure out what caused this? Running into the same issue using Ember 3.4.4. Application works fine locally, but breaks once deployed. Unfortunately too far away from 3.12 to just quickly update.

**Comment by chunyinli** [comment]
Running into the same issue in Ember 3.6.1.

**Comment by rwjblue** [comment]
3.6 is quite old, as is this issue! 😸 

It is highly unlikely that we will backport any fixes to 3.6, and since it seems that at least 3.12 doesn't have the bug your best bet would be to upgrade.
